### PR TITLE
LPS-129260 Crypto Hash OSGi integration

### PR DESCRIPTION
**Pull request by csierra**
/cc @arthurchan35

Here is the OSGi integration part for the Crypto Hash.
I have introduced the Apache Aries Component DSL because it allows us to provide the features we need without having to significantly alter the existing classes (just add a couple of annotations to the existing factories to make them components)

Features are:
* CryptoHashProviderFactories are tracked and registered in the registry. 
* If the factories have an associated configuration.pid the integration will track configurations for those pids. 
* If a configuration for that pid exists it will register a `CryptoHashGenerator` using that configuration.
* Registered `CryptoHashGenerator` services will bear the properties on the service together with the factory name to allow filtering by hashing properties.
* The `CryptoHashVerifier` will work without any existing configuration as long as the corresponding provider is registered. This allows to verify existing hashes that have been generated with weaker configurations and generate new hashes with stronger ones. 
* The factories are tracked by name, making it possible to replace an existing factory with a different implementation with a higher ranking. Regardless of the configuration pid the same factory needs to be used for generation and verification. 
* Configurations are created scoped by company, so services that are going to make use of this framework may make use of a scoped service tracker. 

Carlos.

### CODE REVIEW

File: modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bcrypt-crypto-hash-provider-configuration-name=BCrypt Crypto Hash Provider Configuration
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

File: modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+message-digest-algorithm=Message Digest Algorithm
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

File: modules/apps/portal-crypto-hash/portal-crypto-hash-spi/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -0,0 +1,2 @@
+crypto-hash-provider-configuration-name=Hash Provider Configuration Name
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-spi/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by csierra** [comment]
ci:test:sf

**Comment by csierra** [comment]
ci:test:relevant

**Comment by csierra** [comment]
I don't know why the tests fail... they work locally :man_shrugging:

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by csierra** [comment]
ci:test:relevant

**Comment by csierra** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
Will resend this pull to Stian.

**Pull request closed without merging**